src/main/scala/mesosphere/marathon/core/storage/simple/PersistenceStore.scala

```diff
@@ -0,0 +1,124 @@
+package mesosphere.marathon
+package core.storage.simple
+
+import akka.stream.scaladsl.{Flow, Source}
+import akka.util.ByteString
+import akka.{Done, NotUsed}
+import mesosphere.marathon.core.storage.simple.PersistenceStore.{Node, StoreOp}
+
+import scala.concurrent.Future
+import scala.util.Try
+
+/**
+  * An interface for a simple persistence store abstraction for the underlying Zookeeper store. Supported are basic
+  * create, read, update, delete and check methods along with transaction operations and Zookeeper specific children
+  * and sync methods.
+  */
+trait PersistenceStore {
+
+  /**
+    * A Flow for saving nodes to the store. It takes a stream of nodes and returns a stream of node keys
+    * that were successfully stored.
+    *
+    * By default persistent nodes with empty ACLs are created. If the path is nested parent nodes will be also
+    * created if they don't already exist. If there is already a node with the same path a [[org.apache.zookeeper.KeeperException.NodeExistsException]]
+    * is thrown.
+    *
+    * @return
+    */
+  def create: Flow[Node, String, NotUsed]
+  def create(node: Node): Source[String, NotUsed] = Source.single(node).via(create)
+  def create(nodes: Seq[Node]): Source[String, NotUsed] = Source.fromIterator(() => nodes.iterator).via(create)
+
+  /**
+    * A Flow for reading nodes from the store. It takes a stream of node paths and returns a stream of Try[Node] elements.
+    * It's a Success[Node] for existing nodes or Failure(e) for non-existing nodes where e is an instance of
+    * [[org.apache.zookeeper.KeeperException.NoNodeException]]. The exception contains the path to the node which
+    * simplifies handling failed reads. For other exceptions stream is completed with a failure.
+    *
+    * @return
+    */
+  def read: Flow[String, Try[Node], NotUsed]
```

Comment:
Does this read the root? Should this be private? I'm confused.

Reply:
Same as with creating, the flow is `Path -> Node` e.g. it takes a stream of paths and returns a stream of `Try[Node]` (It's `Try` in case the node does not exist)

Reply:
maybe it would be more apparent if we aliased the String type here? I mean e.g. via `type NodePath = String` or we could aliased that in the Node class...